internal/exec/stages/disks/luks.go

```diff
@@ -294,36 +300,53 @@ func (s *stage) createLuks(config types.Config) error {
 			// pass the device to clevis. We have to loop each device as
 			// the devices could be on different NICs that haven't come
 			// up yet.
+
+			// List of tang servers proven to be reachable.
+			tangServersWithoutAdv := 0
 			for _, tang := range luks.Clevis.Tang {
 				u, err := url.Parse(tang.URL)
 				if err != nil {
 					return fmt.Errorf("parsing tang URL: %v", err)
 				}
 				u.Path = path.Join(u.Path, "adv")
 				_, err = s.Fetcher.FetchToBuffer(*u, resource.FetchOptions{})
-				if err != nil {
-					return fmt.Errorf("fetching tang advertisement: %v", err)
+				if util.NilOrEmpty(tang.Advertisement) {
+					tangServersWithoutAdv++
+					if err != nil {
+						return fmt.Errorf("fetching tang advertisement: %v", err)
+					}
 				}
 			}
 
+			// lets bind our device
 			if _, err := s.Logger.LogCmd(
 				exec.Command(distro.ClevisCmd(), "luks", "bind", "-f", "-k", keyFilePath, "-d", devAlias, pin, config), "Clevis bind",
 			); err != nil {
 				return fmt.Errorf("binding clevis device: %v", err)
 			}
-
-			// close & re-open Clevis devices to make sure that we can unlock them
-			if _, err := s.Logger.LogCmd(
-				exec.Command(distro.CryptsetupCmd(), "luksClose", luks.Name),
-				"closing clevis luks device %v", luks.Name,
-			); err != nil {
-				return fmt.Errorf("closing luks device: %v", err)
+			intTpm2 := 0
+			if luks.Clevis.Tpm2 != nil && *luks.Clevis.Tpm2 {
+				intTpm2 = 1
 			}
-			if _, err := s.Logger.LogCmd(
-				exec.Command(distro.ClevisCmd(), "luks", "unlock", "-d", devAlias, "-n", luks.Name),
-				"reopening clevis luks device %s", luks.Name,
-			); err != nil {
-				return fmt.Errorf("reopening luks device %s: %v", luks.Name, err)
+			threshold := 1
+			if luks.Clevis.Threshold != nil {
```

Comment:
Optional: this calculation could be hoisted and then shared with the `Clevis` struct initialization above.

Reply:
Yeah that would make more sense. I think due to the time crunch and iterations I will be postponing that change.